torch/csrc/cuda/comm.cpp

```diff
@@ -131,8 +152,9 @@ tensor_list2d broadcast_coalesced(TensorList tensors, IntArrayRef devices, size_
         for (auto & t : utils::unflatten_sparse_tensors(inds, vals, chunk.tensors)) {
           // See NOTE [ Version Counter in comm.*_coalesced ]
           AT_ASSERT(t.is_variable());
-          Variable var = t;
-          device_outputs.push_back(make_variable(var.tensor_data(), false));
+          Variable var = as_variable_ref(t).variable_data();
```

Comment:
We specifically want to use `variable_data` instead of `detach` here, because the comment above this function in `NOTE [ Version Counter in comm.*_coalesced ]` mentions:
```cpp
// We thus re-wrap these Variables after broadcasting (i.e., effetively doing
// what is equivalent to .data in Python), and give them individual version
// counters. ...
```
`variable_data` is the strict equivalent of .data in Python, which creates new version counter for the returned tensor, and is exactly what we want.